hydromt/data_adapter_v1/geodataframe_adapter.py

```diff
@@ -0,0 +1,313 @@
+"""The GeoDataFrame adapter performs transformations on GeoDataFrames."""
+from datetime import datetime
+from logging import Logger, getLogger
+from os.path import basename, splitext
+
+import geopandas as gpd
+import numpy as np
+import pyproj
+from pyproj import CRS
+from pyproj.exceptions import CRSError
+from pystac import Asset as StacAsset
+from pystac import Catalog as StacCatalog
+from pystac import Item as StacItem
+from pystac import MediaType
+
+from hydromt.data_sources.geodataframe_data_source import GeoDataFrameDataSource
+from hydromt.gis_utils import filter_gdf, parse_geom_bbox_buffer
+from hydromt.nodata import NoDataStrategy, _exec_nodata_strat
+from hydromt.typing import GEOM_TYPES, ErrorHandleMethod, TotalBounds
+
+from .data_adapter_base import DataAdapterBase
+
+logger: Logger = getLogger(__name__)
+
+
+class GeoDataFrameAdapter(DataAdapterBase):
+    """The GeoDataFrameAdapter performs transformations on GeoDataFrames."""
+
+    source: GeoDataFrameDataSource
+    used: bool = False
+
+    def get_data(
+        self,
+        bbox: list[float] | None = None,
+        mask: gpd.GeoDataFrame | None = None,
+        buffer: float = 0.0,
+        variables: list[str] | None = None,
+        predicate: str = "intersects",
+        handle_nodata: NoDataStrategy = NoDataStrategy.RAISE,
+        logger: Logger | None = None,
```

Comment:
The `| None` syntax doesn't work for python 3.9. Until we drop that you'll have to use `Optional` from the typing library